composeApp/src/commonMain/kotlin/App.kt

```diff
@@ -0,0 +1,80 @@
+import market.MarketApp
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.material.MaterialTheme
+import androidx.compose.material.Surface
+import androidx.compose.ui.Modifier
+import androidx.compose.material.Button
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.MutableState
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.rememberCoroutineScope
+import dev.icerock.moko.permissions.DeniedAlwaysException
+import dev.icerock.moko.permissions.DeniedException
+import dev.icerock.moko.permissions.Permission
+import dev.icerock.moko.permissions.PermissionsController
+import dev.icerock.moko.permissions.compose.BindEffect
+import dev.icerock.moko.permissions.compose.rememberPermissionsControllerFactory
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.launch
+
+internal val RequiredPermissions = arrayOf(
+    Permission.LOCATION,
+    Permission.COARSE_LOCATION,
+    Permission.BLUETOOTH_SCAN,
+)
+
+suspend fun checkPermissions(permissionsController: PermissionsController): Boolean {
+    return RequiredPermissions.all { permission -> permissionsController.isPermissionGranted(permission) }
+}
+
+// TODO: handle denied exceptions properly (redirect user to settings etc.)
+// should also check if bluetooth is open
+suspend fun askPermission(permissionsController: PermissionsController, permission: Permission) {
+    try {
+        permissionsController.providePermission(permission)
+    } catch(deniedAlways: DeniedAlwaysException) {
+        println("$permission is always denied")
+    } catch(denied: DeniedException) {
+        println("$permission is denied")
+    }
+}
+
+@Composable
+fun App() {
```

Comment:
App naming is misleading in both different example screens and the main entry point for the Compose.